src/lib/dai.c

```diff
@@ -136,26 +137,56 @@ static inline const struct dai_type_info *dai_find_type(uint32_t type)
 	return NULL;
 }
 
-struct dai *dai_get(uint32_t type, uint32_t index, uint32_t flags)
+#ifdef __ZEPHYR__
+static struct dai *dai_get_zephyr(const struct dai_type_info *dti, uint32_t type, uint32_t index)
```

Comment:
For minimized `#ifdef __ZEPHYR__` I would have the `dai_get(uint32_t type, uint32_t index, uint32_t flags)` ifdefed.
Or use the same name for the sub dai_get / dai_put OS dependent implementation, but the naming can get funny...
`os_dai_get` / `dai_get_from_os` / ...